lib/apiservers/engine/backends/archive.go

```diff
@@ -461,20 +461,32 @@ func (wm *ArchiveStreamWriterMap) WriterForAsset(proxy proxy.VicArchiveProxy, ci
 }
 
 // Close visits all the archive writer in the trie and closes the actual io.WritCloser
-func (wm *ArchiveStreamWriterMap) Close() {
+func (wm *ArchiveStreamWriterMap) Close() error {
 	defer trace.End(trace.Begin(""))
 
+	numWriter := 0
 	closeStream := func(prefix patricia.Prefix, item patricia.Item) error {
 		if aw, ok := item.(*ArchiveWriter); ok && aw.writer != nil {
-			aw.writer.Close()
 			aw.tarWriter.Close()
-			aw.writer = nil
+			aw.writer.Close()
 			aw.tarWriter = nil
+			aw.writer = nil
+			numWriter++
 		}
 		return nil
 	}
 
 	wm.prefixTrie.Visit(closeStream)
+
+	var err error
+	// wait for all the streams to finish
+	for i := 0; i < numWriter; i++ {
```

Comment:
I am not really sure I really understand what's going on here not seeing all pieces of the code, however, I would consider doing this:
1. Why would you even need to push a nil into errchan? Just push errors only.
2. To wait all writers to complete their job, I would consider using sync.WaitGroup, it will be more clear than depend on the channel content. Also, I think errchan has to be closed as soon all jobs are done, thus you can just iterate over channel using ```for e := range errchan``` loop until you get the last value if that one is needed. However, would definitely consider logging all errors if there are many.